### PR TITLE
Revise generate_buildjl.jl

### PR DESCRIPTION
**Pull request by thofma**
Yes, I am still using this.

### CODE REVIEW

**Comment by thofma** [comment]
Should be good to go.

File: generate_buildjl.jl [hunk 1]
```diff
@@ -87,10 +87,42 @@ function _repr(p::Product)
...
+     else
+         error("Platform $platform not known")
+     end
+end
+
 # `Pkg.CompilerABI` has diverged from `BinaryProvider.CompilerABI`.
 # This function fixes common differences.
 function _pkey(platform)
-    pkey = string(platform_key_abi(platform))
+    pkey = string(compat_string_of_platform_key_abi(platform))
```
Comment (giordano) [inline generate_buildjl.jl:125]: Isn't this
```suggestion
    pkey = string(parse(Platform, platform))
```
?

Comment (thofma) [reply generate_buildjl.jl:125]: The platforms are nowadays printed differently and the output cannot be parsed by julia. Why is your version different to what was there before?

Comment (giordano) [reply generate_buildjl.jl:125]: Then you want
```suggestion
    pkey = repr(parse(Platform, platform))
```

Comment (giordano) [reply generate_buildjl.jl:125]: Oh, I see, this is to get the _old_ representation

Comment (thofma) [reply generate_buildjl.jl:125]: Indeed. I took the lazy approach.

**Comment by giordano** [comment]
Thanks!

**Pull request merged**